tests/expected/function-contract/fail_on_two.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// kani-flags: -Zfunction-contracts
+
+#[kani::ensures(result < 3)]
+fn fail_on_two(i: i32) -> i32 {
+    match i {
+        0 => fail_on_two(i + 1),
+        1 => 2,
+        _ => unreachable!("fail on two"),
```

Comment:
Would it make more sense to make this test fail the postcondition instead via a recursive call, e.g.:
```rust
        2 => fail_on_two(i + 1),
        _ => 3,
```

Reply:
I've added one so now we do both